Sources/VRMSceneKit/Extensions/SCNNode+UnityTransform.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  SCNNode+UnityTransform.swift
+//  VRMSceneKit
+//
+//  Created by Tomoya Hirano on 2019/12/28.
+//  Copyright © 2019 tattn. All rights reserved.
+//
+
+import SceneKit
+
+public protocol UnityTransformCompatible {
+    associatedtype CompatibleType
+    
+    var utx: CompatibleType { get }
+}
+
+public final class UnityTransform<Base> {
+    let base: Base
+    public init(_ base: Base) {
+        self.base = base
+    }
+}
+
+public extension UnityTransformCompatible {
+    var utx: UnityTransform<Self> {
+        return UnityTransform(self)
+    }
+}
+
+extension SCNNode: UnityTransformCompatible {}
+
+extension UnityTransform where Base == SCNNode {
+    func transformPoint(_ position: SCNVector3) -> SCNVector3 {
+        base.convertPosition(position, to: base)
+    }
+    
+    func inverseTransformPoint(_ position: SCNVector3) -> SCNVector3 {
+        base.convertPosition(position, from: base)
+    }
+    
+    var localRotation: SCNQuaternion {
+        get { base.orientation }
+        set { base.orientation = newValue }
+    }
+    
+    var position: SCNVector3 {
+        get { base.worldPosition }
+        set { base.worldPosition = newValue }
+    }
+    
+    var localPosition: SCNVector3 {
+        get { base.position }
+        set { base.position = newValue }
+    }
+    
+    var rotation: SCNQuaternion {
+        get { base.worldOrientation }
+        set { base.worldOrientation = newValue }
+    }
+    
+    var childCount: Int {
+        base.childNodes.count
+    }
+    
+    var worldToLocalMatrix: SCNMatrix4 {
+        base.transform
+    }
+    
+    var lossyScale: SCNVector3 {
+        base.scale
+    }
+}
+
+
+extension SCNNode: Sequence {
+    public func makeIterator() -> ChildNodeIterator {
+        SCNNode.ChildNodeIterator(self)
```

Comment:
Is that the same meaning as the following?:
```swift
extension SCNNode: Sequence {
    public func makeIterator() -> IndexingIterator<[SCNNode]> {
        childNodes.makeIterator()
    }
}
```

Reply:
I tried write code using notation that is original code.
Because it makes easier to follow original code.

Unity-Transform has iterator that looping child nodes.
I implemented for use unity-transform looping in swift.
https://qiita.com/noppefoxwolf/items/295afa6efd6bf5149c59

But this merit is very little.
I'll fix by using `childNodes`.

Reply:
I got it 👍 Thank you for your explanation 🙌